include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -1118,4 +1118,38 @@ def ZOp : OneTargetOp<"z", [Hermitian]> {
   }];
 }
 
+def UnitaryOp : QuakeOp<"unitary", [AttrSizedOperandSegments]> {
```

Comment:
Adding this for reference.

Generally, adding a new op to either of our dialects will require the following testing:

- [ ] adding it to the roundtrip test
- [ ] adding specific cudaq-opt tests that test the ways it is deployed in passes
- [ ] adding a C++ test to make sure the bridge can/does generate the new op
- [ ] adding a backend test to make sure the Quake/CC is lowered to QIR and/or QIR base profile, etc. as expected